src/components/Approval/Approval.tsx

```diff
@@ -0,0 +1,173 @@
+  const symbol = sellAsset.currency?.symbol
+
+  const approve = async () => {
+    if (!wallet) return
+    const txId = await approveInfinite(wallet)
+    if (!txId) return
+    setApprovalTxId(txId)
+    const interval = setInterval(async () => {
+      const approvalNeeded = await checkApprovalNeeded(wallet)
+      if (approvalNeeded) return
+      approvalInterval.current && clearInterval(approvalInterval.current)
+      if (!sellAsset.amount) return
+      if (!quote) return
+      const result = await buildQuoteTx({
+        wallet,
+        sellAsset: quote?.sellAsset,
+        buyAsset: quote?.buyAsset,
+        amount: sellAsset?.amount
+      })
+
+      if (result?.success) {
+        history.push({ pathname: '/trade/confirm', state: { fiatRate } })
+      } else {
+        history.push('/trade/input')
+      }
+    }, 5000)
+    approvalInterval.current = interval
+  }
+
+  useEffect(() => {
+    // TODO: (ryankk) fix errors to reflect correct attribute
+    const error = errors?.quote?.rate ?? null
+    if (error) history.push('/trade/input')
+  }, [errors, history])
+
+  return (
+    <SlideTransition>
+      <Flex
+        justifyContent='center'
+        alignItems='center'
+        flexDirection='column'
+        width='full'
+        as='form'
+        onSubmit={handleSubmit(approve)}
+      >
+        <CountdownCircleTimer
+          isPlaying={!!approvalTxId || !!isSubmitting}
+          size={90}
+          strokeWidth={6}
+          trailColor={theme.colors.whiteAlpha[500]}
+          duration={60}
+          colors={[
+            [theme.colors.blue[500], 0.4],
+            [theme.colors.blue[500], 0.4]
+          ]}
+          onComplete={() => {
+            return [true, 0]
+          }}
+        >
+          <Image
+            src={sellAsset.currency?.icon}
+            boxSize='60px'
+            fallback={<SkeletonCircle boxSize='60px' />}
+          />
+        </CountdownCircleTimer>
+        <Text
+          my={2}
+          fontSize='lg'
+          fontWeight='bold'
+          textAlign='center'
+          translation={['trade.approveAsset', { symbol }]}
+        />
+        <Text color='gray.500' translation={['trade.needPermission', { symbol }]} />
+        <Link
+          isExternal
+          color='blue.500'
+          href='https://shapeshift.zendesk.com/hc/en-us/articles/360018501700'
+        >
+          <Text color='blue.500' translation='trade.whyNeedThis' />
+        </Link>
+        <Divider my={4} />
+        <Flex flexDirection='column' width='full'>
+          {approvalTxId && sellAsset.currency?.explorerTxLink && (
+            <Row>
+              <Row.Label>
+                <Text translation={['trade.approvingAsset', { symbol }]} />
+              </Row.Label>
+              <Row.Value>
+                <Link
+                  isExternal
+                  color='blue.500'
+                  href={`${sellAsset.currency?.explorerTxLink}${approvalTxId}`}
```

Comment:
We really should have helper functions that create explorer links given a link template and a value.